mbed_devices/_internal/windows/device_instance_id.py

```diff
@@ -0,0 +1,56 @@
+"""Utility in charge of finding the instance ID of a device."""
+import win32con
+import win32api
+from mbed_devices._internal.exceptions import SystemException
+import logging
+
+from typing import Optional, Any
+
+logger = logging.getLogger(__name__)
+
+
+class RegKey(object):
+    """Context manager in charge of opening and closing registry keys."""
+
+    def __init__(self, sub_registry_key: str) -> None:
+        """Initialiser."""
+        access = win32con.KEY_READ | win32con.KEY_ENUMERATE_SUB_KEYS | win32con.KEY_QUERY_VALUE
+        try:
+            self._hkey = win32api.RegOpenKey(win32con.HKEY_LOCAL_MACHINE, sub_registry_key, 0, access)
+        except win32api.error as e:
+            raise SystemException(f"Could not read key [{sub_registry_key}] in the registry: {e}")
+
+    def __enter__(self) -> Any:
+        """Actions on entry."""
+        return self._hkey
+
+    def __exit__(self, type: Any, value: Any, traceback: Any) -> None:
+        """Actions on exit."""
+        win32api.RegCloseKey(self._hkey)
+        self._hkey.close()
+
+
+def getChildrenInstanceId(pnpid: str) -> Optional[str]:
```

Comment:
Is there a reason this function name is camel case? We should use snake case by default.

Reply:
oops, my bad.